### PR TITLE
Add `PyTuple::to_list`

### PR DESCRIPTION
**Pull request by davidhewitt**
Companion to #3043. I've included benchmarks which suggests that this method is indeed about 25% faster on my machine.

### CODE REVIEW

File: src/types/tuple.rs [hunk 1]
```diff
@@ -213,6 +214,15 @@ impl PyTuple {
...
     }
+
+    /// Return a new list containing the contents of this tuple; equivalent to the Python expression `list(tuple)`.
+    ///
+    /// This method is equivalent to `self.as_sequence().list()` and faster than `PyList::new(py, self)`.
+    pub fn to_list(&self) -> &PyList {
+        self.as_sequence()
+            .list()
+            .expect("failed to convert tuple to list")
+    }
```
Comment (mejrs) [inline src/types/tuple.rs:225]: (I think) you can do  `PyList::new(py, self.as_slice())` on `not(Py_LIMITED_API)`

Comment (davidhewitt) [reply src/types/tuple.rs:225]: Oh great idea, I'll try that when I get a free moment and see how it performs...

Comment (davidhewitt) [reply src/types/tuple.rs:225]: Hmm unfortunately that performs very similarly to `PyList::new(py, self)` (i.e. slower than going through the sequence API), so the bottleneck must be in constructing the list on the Rust side. An investigation for another time, perhaps...

**Comment by davidhewitt** [comment]
Based on #3043 having already gone in, I'm just going to proceed to merge this so I can include it in a rebased 0.18.2.

bors r+

**Pull request merged**